### PR TITLE
InsertNode with option InsertInNewLine is inserting the new node incorrectly when target is a table cell

### PR DESCRIPTION
**Pull request by BryanValverdeU**
**Describe the bug**
Using InsertNode Editor API inside of a table and setting the Insert Option, InsertInNewLine to true, it is inserting the new element incorrectly.

**Code Change**
If inside of a cell add a Div so we can insert the content in a new line, instead of inserting the content next to the cell

**Before**
image

**After**
InsertNode1

### CODE REVIEW

File: packages/roosterjs-editor-core/lib/coreApi/insertNode.ts [hunk 1]
```diff
@@ -179,3 +178,23 @@ export const insertNode: InsertNode = (core: EditorCore, node: Node, option: Ins
...
+    let tempPos = new Position(blockElement.getEndNode(), PositionType.After);
+    if (safeInstanceOf(tempPos.node, 'HTMLTableRowElement')) {
+        const div = core.contentDiv.ownerDocument.createElement('div');
+        div.id = 'newLineInTd';
+        insertNode(core, div, {
+            position: ContentPosition.SelectionStart,
+            insertOnNewLine: false,
+            updateCursor: true,
+            replaceSelection: true,
+        });
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-core/lib/coreApi/insertNode.ts:191]: Inside insertNode() function, can we avoid calling it recursively? Can we just directly insert the node using some DOM API such as insertBefore or appendChild?

File: packages/roosterjs-editor-core/lib/coreApi/insertNode.ts [hunk 2]
```diff
@@ -179,3 +178,23 @@ export const insertNode: InsertNode = (core: EditorCore, node: Node, option: Ins
 
     return true;
 };
+function adjustInsertPositionNewLine(blockElement: BlockElement, core: EditorCore, pos: Position) {
+    let tempPos = new Position(blockElement.getEndNode(), PositionType.After);
+    if (safeInstanceOf(tempPos.node, 'HTMLTableRowElement')) {
+        const div = core.contentDiv.ownerDocument.createElement('div');
+        div.id = 'newLineInTd';
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-core/lib/coreApi/insertNode.ts:185]: No need to use an Id, just hold the reference of DIV and it won't change.

File: packages/roosterjs-editor-core/lib/coreApi/insertNode.ts (hunk 1 above)
Comment (BryanValverdeU) [reply packages/roosterjs-editor-core/lib/coreApi/insertNode.ts]: Sounds good will use the Range.insertNode Api to do this.

File: packages/roosterjs-editor-core/lib/coreApi/insertNode.ts [hunk 3]
```diff
@@ -179,3 +179,13 @@ export const insertNode: InsertNode = (core: EditorCore, node: Node, option: Ins
 
     return true;
 };
+function adjustInsertPositionNewLine(blockElement: BlockElement, core: EditorCore, pos: Position) {
+    let tempPos = new Position(blockElement.getEndNode(), PositionType.After);
+    if (safeInstanceOf(tempPos.node, 'HTMLTableRowElement')) {
+        const div = core.contentDiv.ownerDocument.createElement('div');
+        const range = createRange(pos);
+        range.insertNode(div);
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-core/lib/coreApi/insertNode.ts:187]: Wait, is this going to insert a DIV after a TR?

Can you give me an example including:
- HTML before insert
- The position
- HTML after insert

Comment (BryanValverdeU) [reply packages/roosterjs-editor-core/lib/coreApi/insertNode.ts:187]: No, it is going to insert a Div inside of a TD. To treat it as a new Line inside of a table cell

Comment (BryanValverdeU) [reply packages/roosterjs-editor-core/lib/coreApi/insertNode.ts:187]: Before Insert 
image

Position
image

After Insert
image

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**